File: src/utils/yaml.ts
import { parse, stringify } from 'yaml';
import { readText, writeText } from './fs.js';

export async function readYaml<T = unknown>(path: string): Promise<T> {
	const text = await readText(path);
	return parse(text) as T;
}

export async function writeYaml(path: string, data: unknown): Promise<void> {
	const text = stringify(data, { lineWidth: 120 });
	await writeText(path, text);
}

export { parse as parseYaml, stringify as stringifyYaml } from 'yaml';

File: internal/util/yaml.go
package util

import (
	"gopkg.in/yaml.v3"
)

// ReadYAML reads and parses a YAML file.
func ReadYAML(path string, v any) error {
	data, err := ReadText(path)
	if err != nil {
		return err
	}
	return yaml.Unmarshal([]byte(data), v)
}

// WriteYAML writes a value as YAML to a file.
func WriteYAML(path string, v any) error {
	data, err := yaml.Marshal(v)
	if err != nil {
		return err
	}
	return WriteText(path, string(data))
}
